src/mahoji/commands/runecraft.ts

```diff
@@ -127,6 +139,27 @@ export const runecraftCommand: OSBMahojiCommand = {
 		const numEssenceOwned = bank.amount('Pure essence');
 		const daeyaltEssenceOwned = bank.amount('Daeyalt essence');
 
+		const warpedRunes = new Set(['air', 'mind', 'water', 'earth', 'fire', 'body']);
+		const twistedRunes = new Set(['mist', 'dust', 'mud', 'smoke', 'steam', 'lava', 'cosmic', 'chaos', 'sunfire']);
+		const mangledRunes = new Set(['nature', 'law', 'astral', 'death']);
+		const scarredRunes = new Set(['blood', 'soul', 'wrath']);
+
+		let extractsOwned = 0;
+		let extractType = '';
+		if (warpedRunes.has(rune)) {
+			extractsOwned = bank.amount('Warped extract');
+			extractType = 'Warped extract';
+		} else if (twistedRunes.has(rune)) {
+			extractsOwned = bank.amount('Twisted extract');
+			extractType = 'Twisted extract';
+		} else if (mangledRunes.has(rune)) {
+			extractsOwned = bank.amount('Mangled extract');
+			extractType = 'Mangled extract';
+		} else if (scarredRunes.has(rune)) {
+			extractsOwned = bank.amount('Scarred extract');
+			extractType = 'Scarred extract';
+		}
```

Comment:
This should probably be something like (defined outside this function):

```ts
const types = [
{
item: getOSItem("Warped extract"),
runes: new Set(),
}
];
```

so you dont need the if elses's